diff --git a/tools/server/server.cpp b/tools/server/server.cpp
--- a/llama.cpp/tools/server/server.cpp
+++ b/llama.cpp/tools/server/server.cpp
@@ -15,6 +15,10 @@
 #include <windows.h>
 #endif

+#ifdef COSMOCC
+#include <cosmo.h>
+#endif
+
 static std::function<void(int)> shutdown_handler;
 static std::atomic_flag is_terminating = ATOMIC_FLAG_INIT;

@@ -66,6 +70,10 @@ static server_http_context::handler_t ex_wrapper(server_http_context::handler_t
 }

 int main(int argc, char ** argv, char ** envp) {
+    #ifdef COSMOCC
+    argc = cosmo_args("/zip/.args", &argv);
+    #endif
+
     // own arguments required by this example
     common_params params;

